pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -88,6 +89,7 @@
     protected final ConcurrentLinkedQueue<CompletableFuture<Message<T>>> pendingReceives;
     protected final int maxReceiverQueueSize;
     private volatile int currentReceiverQueueSize;
+    protected final AtomicInteger messageListenerQueueSize = new AtomicInteger(0);
```

Comment:
Suggest to use `AtomicIntegerFieldUpdater`

Reply:
Done